data/src/main/java/me/androidbox/data/worker_manager/UploadEventImp.kt

```diff
@@ -23,22 +24,25 @@ class UploadEventImp @Inject constructor(
         const val ERROR = "error"
     }
 
-    override fun upload(event: Event, isEditMode: Boolean) {
+    override fun upload(event: Event, updateModeType: UpdateModeType) {
         /** Serialize the event to json */
-        val eventRequestJson = if(isEditMode) {
-            /** request for update eventRequestDto */
-            val jsonAdapter = moshi.adapter(EventUpdateRequestDto::class.java)
-            jsonAdapter.toJson(event.toUpdateEventDto())
-        }
-        else {
-            /** request for Create eventRequestDto */
-            val jsonAdapter = moshi.adapter(EventCreateRequestDto::class.java)
-            jsonAdapter.toJson(event.toCreateEventDto())
+        val eventRequestJson: String = when(updateModeType) {
+            UpdateModeType.CREATE -> {
+                /** request for Create eventRequestDto */
+                val jsonAdapter = moshi.adapter(EventCreateRequestDto::class.java)
+                jsonAdapter.toJson(event.toCreateEventDto())
```

Comment:
You don't really need to create the adapter separately here if you don't reuse it